command/ssh.go

```diff
@@ -808,12 +809,23 @@ func (c *SSHCommand) parseSSHCommand(args []string) (hostname string, username s
 			continue
 		}
 
+		// If -l has been specified then this is our user name
+		if arg == "-l" {
+			username = i
+			continue
+		}
+
 		// this is an ssh option, lets see if User or Port have been set and use it
 		if arg == "-o" {
 			split := strings.Split(i, "=")
 			key := split[0]
 			// Incase the value contains = signs we want to get all of them
-			value := strings.Join(split[1:], " ")
+			value := strings.Join(split[1:], "=")
```

Comment:
I agree this was wrong before.  Instead of this fix, I would rather you replace the `strings.Split` above with a call to `strings.SplitN` with a value of n=2.  Then we won't need to join anything.